### PR TITLE
chore: produce go bindings

### PR DESCRIPTION
**Pull request by eladb**
Adds a `go.moduleName` entry to all package.json files which will cause jsii-pacmak to produce go bindings for the AWS CDK and place them under the `dist/go` directory during packaging.

Resolves https://github.com/aws/jsii/issues/2611

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by RomainMuller — COMMENTED** [review]
I don't think it's good/expected that *all* of those use the exact same module name...?

**Comment by eladb** [comment]
> I don't think it's good/expected that all of those use the exact same module name...?

Yes, this is intentional. `moduleName` is actually just the prefix. The actual module name is derived from the npm module name.

File: tools/pkglint/lib/rules.ts [hunk 1]
```diff
@@ -695,6 +695,21 @@ export class JSIIPythonTarget extends ValidationRule {
...
+
+export class JSIIGolangTarget extends ValidationRule {
+  public readonly name = 'jsii/go';
+
+  public validate(pkg: PackageJson): void {
+    if (!isJSII(pkg)) { return; }
+
+    const moduleName = cdkModuleName(pkg.json.name);
+
+    // See: https://github.com/aws/jsii/blob/master/docs/configuration.md#configuring-python
```
Comment (RomainMuller) [inline tools/pkglint/lib/rules.ts:707]: Bad link - also, the existing documentation does not mention any configuration... Whoop whoop.

```suggestion
    // See: https://aws.github.io/jsii/user-guides/lib-author/configuration/targets/go/
```

File: tools/pkglint/lib/rules.ts [hunk 2]
```diff
@@ -869,6 +884,7 @@ function cdkModuleName(name: string) {
       distName: `aws-cdk.${pythonName}`,
       module: `aws_cdk.${pythonName.replace(/-/g, '_')}`,
     },
+    goRootModuleName: 'github.com/aws/aws-cdk-go',
```
Comment (RomainMuller) [inline tools/pkglint/lib/rules.ts:887]: Possible alternative (take it or leave it):

```suggestion
    goDistributionRepository: 'github.com/aws/aws-cdk-go',
```

File: tools/pkglint/lib/rules.ts (hunk 1 above)
Comment (iliapolo) [inline tools/pkglint/lib/rules.ts:707]: Seems outdated. 

```suggestion
    // See: https://github.com/aws/jsii/blob/main/gh-pages/content/user-guides/lib-author/configuration/targets/go.md
```

**Review by RomainMuller — APPROVED** [review]
Provisional approval (feel free to remove `pr/do-not-merge` when you've got all the feedback you asked for!)

File: tools/pkglint/lib/rules.ts (hunk 1 above)
Comment (RomainMuller) [reply tools/pkglint/lib/rules.ts:707]: My suggestion is better 🙃

**Review by iliapolo — APPROVED** [review]

File: tools/pkglint/lib/rules.ts (hunk 2 above)
Comment (eladb) [reply tools/pkglint/lib/rules.ts:887]: Changing to `goRepositoryName` to align with the proposal in https://github.com/aws/jsii/issues/2632

File: tools/pkglint/lib/rules.ts (hunk 1 above)
Comment (eladb) [reply tools/pkglint/lib/rules.ts:707]: Done

**Pull request merged**